Sources/TurboNavigator.swift

```diff
@@ -1,235 +1,89 @@
+    /// Or, provide a completely custom `WKUIDelegate` implementation.
+    public var webkitUIDelegate: TurboWKUIController? {
+        didSet {
+            session.webView.uiDelegate = webkitUIDelegate
+            modalSession.webView.uiDelegate = webkitUIDelegate
+        }
     }
 
-    /// Provide `Turbo.Session` instances with preconfigured path configurations and delegates.
-    /// Note that TurboNavigationDelegate.controller(_:forProposal:) will no longer be called.
+    /// Default initializer requiring preconfigured `Session` instances.
+    /// User `init(pathConfiguration:delegate)` to only provide a `PathConfiguration`.
     /// - Parameters:
-    ///   - preconfiguredMainSession: a session whose delegate is not `TurboNavigator`
-    ///   - preconfiguredModalSession: a session whose delegate is not `TurboNavigator`
-    ///   - delegate: handle non-routing behavior, like custom error handling
-    ///   - navigationController: optional: override the main navigation stack
-    ///   - modalNavigationController: optional: override the modal navigation stack
-    public init(preconfiguredMainSession: Turbo.Session,
-                preconfiguredModalSession: Turbo.Session,
-                delegate: TurboNavigationDelegate,
-                navigationController: UINavigationController = UINavigationController(),
-                modalNavigationController: UINavigationController = UINavigationController())
-    {
-        self.session = preconfiguredMainSession
-        self.modalSession = preconfiguredModalSession
-        self.navigationController = navigationController
-        self.modalNavigationController = modalNavigationController
+    ///   - session: the main `Session`
+    ///   - modalSession: the `Session` used for the modal navigation controller
+    ///   - delegate: an optional delegate to handle custom view controllers
+    public init(session: Session, modalSession: Session, delegate: TurboNavigatorDelegate? = nil) {
+        self.session = session
+        self.modalSession = modalSession
 
-        self.delegate = delegate
+        self.delegate = delegate ?? navigatorDelegate
+
+        self.session.delegate = self
+        self.modalSession.delegate = self
+
+        // Defer to trigger didSet callback.
+        defer { self.webkitUIDelegate = TurboWKUIController(delegate: self) }
     }
 
-    public var rootViewController: UIViewController { navigationController }
-    public let navigationController: UINavigationController
-    public let modalNavigationController: UINavigationController
+    /// Convenience initializer that doesn't require manually creating `Session` instances.
+    /// - Parameters:
+    ///   - pathConfiguration:
+    ///   - delegate: an optional delegate to handle custom view controllers
+    public convenience init(pathConfiguration: PathConfiguration, delegate: TurboNavigatorDelegate? = nil) {
```

Comment:
I really, really like this initializer. I think it's now very clear that the only required dependency for `TurboNavigator` is the path configuration.

Reply:
Building on that… I don't think it is unreasonable for us to provide a default path configuration if one isn't provided. One that handles the required routing, only.